### PR TITLE
Internalizing MetadataUtils

### PR DESCRIPTION
**Pull request by Zruty0**
Contributes to #1500 

Moved most of MetadataUtils to internal, except some public methods to access common metadata.
Removed some methods that are now trivial.

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
>     public static class MetadataUtils

If you internalize MetadataUtils, why it's still public :) ?

---
Refers to: src/Microsoft.ML.Core/Data/MetadataUtils.cs:21 in 612075c. [](commit_id = 612075ca5143d917bdd47eef731999cd01da283e, deletion_comment = False)

File: src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@
 using Newtonsoft.Json.Linq;
 using Microsoft.ML.Runtime.Model.Pfa;
 using System.Collections.Generic;
+using Microsoft.ML.Data;
```
Comment (Zruty0) [inline src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs]: >using [](start = 0, length = 5)

I'm sorting all the usings now, don't do the Latin counting again!

**Comment by Ivanidzo4ka** [comment]
> 

Can you kill it and sort?

---
Refers to: src/Microsoft.ML.Data/Depricated/Instances/HeaderSchema.cs:8 in 612075c. [](commit_id = 612075ca5143d917bdd47eef731999cd01da283e, deletion_comment = False)

File: src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs (hunk 1 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs]: I can do Japanese! Ichi, ni, san, shi ...

---
In reply to: [242338122](https://github.com/dotnet/machinelearning/pull/1893#discussion_r242338122) [](ancestors = 242338122)

**Comment by Zruty0** [comment]
Please look at MetadataUtils first: all the other files are automatically changed to continue compiling with the changes in MetadataUtils.

**Comment by TomFinley** [comment]
>     public static class MetadataUtils

Parts of it. I suspect though that just having this list of string constants doesn't really do much of anyone any good -- if we want conveniences on top of `Schema.Metadata` that's great, but this by itself we might consider just an intermediate stage. But, as an intermeidate stage I think it's fine, we definitely need to change the namespace anyway...

---
In reply to: [448022694](https://github.com/dotnet/machinelearning/pull/1893#issuecomment-448022694) [](ancestors = 448022694,448020032)

---
Refers to: src/Microsoft.ML.Core/Data/MetadataUtils.cs:21 in 612075c. [](commit_id = 612075ca5143d917bdd47eef731999cd01da283e, deletion_comment = False)

**Pull request merged**